cpu/native/periph/random.c

```diff
@@ -40,33 +72,89 @@ int random_read(char *buf, unsigned int num)
         return 0;
     }
 
-    DEBUGF("random_read: writing %ui bytes\n", num);
-    for (unsigned i = 0; i < num; i++) {
-        _native_syscall_enter();
-        buf[i] = (char)real_random();
-        _native_syscall_leave();
+    DEBUG("random_read: writing %u bytes\n", num);
+    switch (_native_rng_mode) {
+        case 0:
+            num = _native_rng_read_hq(buf, num);
+            break;
+        case 1:
+            num = _native_rng_read_det(buf, num);
+        default:
+            DEBUG("random_read: _native_rng_mode is in invalid state %i\n",
+                   _native_rng_mode);
```

Comment:
I'd argue that while it's not meant to be possible there is still no downside in communicating it (meaning that I'm not seeing one :D). An error that is never displayed doesn't hurt, but if for any reason whatsoever `_native_rng_mode` is set to a weird value it'll make debugging that helluvalot faster :hamster: